src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -845,8 +846,11 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
                 return false;
             });
 
-        pass_config->enable<ov::pass::NormalizeL2Decomposition>();
-        pass_config->set_callback<ov::pass::NormalizeL2Decomposition>(
+        // ReduceSum would causes fp16 out of range.
+        // To avoid the out of range, use fp32 nodes for NormalizeL2Decomposition.
+        bool use_fp32_internal_nodes = (infer_precision == ov::element::f16);
+        manager.register_pass<NormalizeL2Decomposition>(use_fp32_internal_nodes);
```

Comment:
ov::pass::NormalizeL2Decomposition is being changed to ov::intel_gpu::NormalizeL2Decomposition, right? For better readability, it is recommended to include the namespace.

Reply:
Updated! Thanks!